### DISCUSSION TITLE
How to reset file field on custom page?

### DISCUSSION THREAD

**Discussion by imknight** [filamentphp/filament#13924] [Help]
### Package

Form builder

### Package Version

v3.2.96

### How can we help you?

I have a form with just an upload file field; I would like to trigger a process once the user finishes uploading the file and resetting the form. I have try a few ways but it seems like the file are still remain there, anything I miss out?

`class Test extends Page implements HasForms, HasActions
{
    use InteractsWithActions;
    use InteractsWithForms;

    public ?array $data = [];

    public $content = false;

    protected static ?string $navigationIcon = "heroicon-o-document-text";

    protected static string $view = "filament.portal.pages.project";

    protected static bool $shouldRegisterNavigation = false;

    public function form(Form $form): Form
    {
        return $form
            ->schema([
                FileUpload::make("media")
                    ->image()
                    ->maxSize(4000)
                    ->imageResizeMode("cover")
                    ->imageResizeTargetHeight("1024")
                    ->imageResizeTargetWidth("1024")
                    ->afterStateUpdated(function ($livewire, $state) {
                        if ($state) {
                            $livewire->sendMedia($state);
                        }
                        //$this->loading = false;
                    })
                    ->hiddenLabel(),
            ])
            ->statePath("data");
    }

    public function sendMedia($state)
    {
        dd($state);
    }
}`

**Comment by leandrocfe** [comment]
try
```php
if ($state) {
    $livewire->sendMedia($state);
    $livewire->data['media'] = null;
}
```

**Reply by imknight** [reply]
nope, try that not working, the file still showing

**Comment by envolute** [comment]
I have the same problem...

I've already tried
$this->data['file'] = null;
$this->form->fill();

nothing!!

But I realized that the problem is not in the input file, it resets, but the "filepond" doesn't clear the listed file. Do you understand??
There must be some way to reset filepond, but I haven't found it.
